integration-app/src/main/java/uk/gov/crowncommercial/esourcing/integration/service/TenderApiService.java

```diff
@@ -161,38 +172,88 @@ public TenderApiService(
+            .block(Duration.ofSeconds(defaultApiTimeout));
+
+    final RfxSetting rfxSetting;
+    if (rfx != null) {
+      if (rfx.getDataList() != null) {
+        if (rfx.getDataList().getRfx() != null) {
+          rfxSetting =
+              rfx.getDataList() != null ? rfx.getDataList().getRfx().get(0).getRfxSetting() : null;
+        } else {
+          throw new IllegalStateException(
+              String.format("No RFx details found for refReferenceCode - %s", eventID));
+        }
+      } else {
+        throw new IllegalStateException(
+            String.format("No RFx dataList found for refReferenceCode - %s", eventID));
+      }
+    } else {
+      throw new IllegalStateException(
+          String.format("No RFx found for refReferenceCode - %s", eventID));
+    }
 
-    logger.info("Invalidating ITT : {}", abandonITT);
+    Integer ownerUserId;
+    if (rfxSetting != null) {
+      ownerUserId = rfxSetting.getOwnerUser().getId();
+    } else {
+      throw new IllegalStateException(
+          String.format("No Owner found for refReferenceCode - %s", eventID));
+    }
 
+    String status = rfxSetting.getStatus();
     OperatorUser operatorUser = new OperatorUser();
-    operatorUser.setId(defaultOwnerUserId);
+    operatorUser.setId(ownerUserId);
 
     RfxInvalidate rfxInvalidate = new RfxInvalidate();
-    rfxInvalidate.setRfxReferenceCode(abandonITT.getRfxReferenceCode());
     rfxInvalidate.setOperatorUser(operatorUser);
+    rfxInvalidate.setRfxReferenceCode(eventID);
+
+    RfxInvalidationResponse jaggaerResponse;
+
+    if (status != null && status.equals("To be Published")) {
+      jaggaerResponse =
+          rfxWorkflowsApi.deleteRFX(rfxInvalidate).block(Duration.ofSeconds(defaultApiTimeout));
+    } else {
+      jaggaerResponse =
+          rfxWorkflowsApi.invalidateRFX(rfxInvalidate).block(Duration.ofSeconds(defaultApiTimeout));
+    }
+    final String finalStatus = (jaggaerResponse == null) ? "" : jaggaerResponse.getFinalStatus();
+
+    String response;
 
-    RfxInvalidationResponse jaggaerResponse = rfxWorkflowsApi.invalidateRFX(rfxInvalidate).block(Duration.ofSeconds(defaultApiTimeout));
+    // Convert Jaggaer termination status to OCDS
+    if (finalStatus != null && finalStatus.equalsIgnoreCase("Invalidated")) {
+      response = "withdrawn";
+    } else if (finalStatus != null && finalStatus.equalsIgnoreCase("Deleted")) {
+      response = "cancelled";
+    } else {
+      response = finalStatus;
+    }
 
     if (jaggaerResponse != null) {
-      response.setReturnMessage(jaggaerResponse.getReturnMessage());
-      response.setRfxReferenceCode(jaggaerResponse.getRfxReferenceCode());
-      response.setFinalStatus(jaggaerResponse.getFinalStatus());
-      return new ResponseEntity<>(response, HttpStatus.OK);
+      if (jaggaerResponse.getReturnMessage() != null
+          && jaggaerResponse.getReturnMessage().contains("rfx can't be invalidate")) {
```

Comment:
I would add comments to explain why this is being checked for like this and not handled in a nicer way.

Reply:
Comment addded